src/NLog/LayoutRenderers/DirectorySeparatorLayoutRenderer.cs

```diff
@@ -0,0 +1,74 @@
+﻿// 
+// Copyright (c) 2004-2019 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.LayoutRenderers
+{
+    using System.IO;
+    using System.Text;
+    using NLog.Internal;
+    using NLog.Config;
+
+    /// <summary>
+    /// The OS dependent directory separator
+    /// </summary>
+    [LayoutRenderer("dir-separator")]
+    [ThreadAgnostic]
+    [ThreadSafe]
+    public class DirectorySeparatorLayoutRenderer : LayoutRenderer, IStringValueRenderer, IRawValue
+    {
+        private static readonly char SeparatorChar;
+        private static readonly string Separator;
+
+        static DirectorySeparatorLayoutRenderer()
+        {
+            SeparatorChar = Path.DirectorySeparatorChar;
+            Separator = SeparatorChar.ToString();
+        }
+
+        /// <inheritdoc/>
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            builder.Append(SeparatorChar);
+        }
+
+        /// <inheritdoc/>
+        string IStringValueRenderer.GetFormattedString(LogEventInfo logEvent) => Separator;
+
+        /// <inheritdoc />
+        public bool TryGetRawValue(LogEventInfo logEvent, out object value)
```

Comment:
Think the raw-value should return the string and not the char.

Reply:
The raw value **is** a char. Why should it be a string?

Reply:
Because that is what people expect. No one uses single char.

Reply:
How does rawvalue work with literallayoutrenderer? Does it return single char when only a single letter?

Reply:
Now, and that because literallayoutrenderer isn't implementing IRawValue

Reply:
anyway, you could use this in char columns in the database as well as strings. But OK, that's not really an use full case.

Reply:
It's char because the underlining type is char. I think that is a good practice?

Reply:
Oh well I guess we will never know since it will always be turned into a literallayoutrenderer :)